### ISSUE TITLE
Evangelion Jo : Missing menu text

### ISSUE THREAD

**Issue by dbz400** [hrydgard/ppsspp#3908]
Just for record. Looks like it is sceFont issue .

![screen00027](https://f.cloud.github.com/assets/3000282/1199536/68e64054-2512-11e3-9432-64ef1455f1d8.jpg)

HLE\sceFont.cpp:877 sceFontGetCharGlyphImage_Clip(09f094dc, 65326, 09ffe710, 0, 0, 254, 30)
HLE\sceFont.cpp:877 sceFontGetCharGlyphImage_Clip(09f094dc, 65332, 09ffe710, 0, 0, 254, 30)
HLE\sceFont.cpp:877 sceFontGetCharGlyphImage_Clip(09f094dc, 65321, 09ffe710, 0, 0, 254, 30)
HLE\sceFont.cpp:877 sceFontGetCharGlyphImage_Clip(09f094dc, 65326, 09ffe710, 0, 0, 254, 30)
HLE\sceFont.cpp:877 sceFontGetCharGlyphImage_Clip(09f094dc, 65333, 09ffe710, 0, 0, 254, 30)
HLE\sceFont.cpp:877 sceFontGetCharGlyphImage_Clip(09f094dc, 65317, 09ffe710, 0, 0, 254, 30)

**Comment by daniel229** [comment]
Still missing text
![1](https://cloud.githubusercontent.com/assets/3481559/12633317/9e6513ea-c5ab-11e5-877f-83c87ef22f67.png)

```
39:29:591 user_main    I[SCEGE]: Common\FramebufferCommon.cpp:412 Creating FBO for 001fe7c0 : 156 x 12 x 0
39:29:593 user_main    I[SCEGE]: Common\FramebufferCommon.cpp:412 Creating FBO for 001fcf00 : 156 x 12 x 0
39:29:594 user_main    I[SCEGE]: Common\FramebufferCommon.cpp:412 Creating FBO for 001fb640 : 156 x 12 x 0
39:29:607 idle0        I[SCEGE]: GLES\Framebuffer.cpp:1896 Decimating FBO for 001fe7c0 (156 x 12 x 0), age 6
39:29:608 idle0        I[SCEGE]: GLES\Framebuffer.cpp:1896 Decimating FBO for 001fcf00 (156 x 12 x 0), age 6
39:29:609 idle0        I[SCEGE]: GLES\Framebuffer.cpp:1896 Decimating FBO for 001fb640 (156 x 12 x 0), age 6
```
if I comment out these,it shows up,still not correct.
```
		/*if (vfb != displayFramebuf_ && vfb != prevDisplayFramebuf_ && vfb != prevPrevDisplayFramebuf_) {
			if (age > FBO_OLD_AGE) {
				INFO_LOG(SCEGE, "Decimating FBO for %08x (%i x %i x %i), age %i", vfb->fb_address, vfb->width, vfb->height, vfb->format, age);
				DestroyFramebuf(vfb);
				vfbs_.erase(vfbs_.begin() + i--);
			}
		}*/
```
![2](https://cloud.githubusercontent.com/assets/3481559/12633318/a2534008-c5ab-11e5-8037-821130eb77af.png)

GE debugger
![3](https://cloud.githubusercontent.com/assets/3481559/12633352/f76fc124-c5ab-11e5-9cf3-874e536e0084.png)
https://gist.github.com/daniel229/58998ba723dc08834dac

**Comment by daniel229** [comment]
Break on 0x041FE7C0,got this function.
https://cloud.githubusercontent.com/assets/3481559/12633684/920ae3b0-c5ae-11e5-8080-fa87a225b250.png

**Comment by unknownbrackets** [comment]
Interesting, this uses memcpy and memset.  Do the changes in #8359 help?

-[Unknown]

**Comment by daniel229** [comment]
No,it deosn't.

**Comment by daniel229** [comment]
Loading savestate from softgpu,look good.
![5](https://cloud.githubusercontent.com/assets/3481559/12638077/3c5856d8-c5d5-11e5-800b-d51f98618d19.png)

**Comment by sum2012** [comment]
v1.2.2-288-g9a11cfb still missing text

**Comment by sum2012** [comment]
v1.2.2-288-g9a11cfb log:
https://gist.github.com/sum2012/42bfe40c7e65645bdf715521fd751d04

Maybe we need imp SysMemUserForUser_D8DE5C1E() ?

**Comment by unknownbrackets** [comment]
Perhaps.  It seems that our implementation was from Jpcsp, and they both just return 0.  It still doesn't work in Jpcsp, right?

Given it's in sysmem, maybe SysMemUserForUser_D8DE5C1E is an allocation function.  Can you change:

```c++
ERROR_LOG(SCEKERNEL,"UNIMPL SysMemUserForUser_D8DE5C1E()");
```

To:

```c++
ERROR_LOG(SCEKERNEL,"UNIMPL SysMemUserForUser_D8DE5C1E(%08x, %08x, %08x, %08x, %08x, %08x)", PARAM(0), PARAM(1), PARAM(2), PARAM(3), PARAM(4), PARAM(5));
```

According to Google, some people claim this function is `sceKernelSafetyCheck0`.  If so, maybe it's unlikely it would impact functionality... not sure what it does or is meant to do.

Is it possible to log that func via JpcspTrace?

-[Unknown]

**Comment by sum2012** [comment]
Modify ppsspp log:
https://gist.github.com/sum2012/01991fc9a5a79d6e654af86f58034401

JPCSPTrace log:
https://gist.github.com/sum2012/d982a3af6e66de346792436e083400b2

JPCSP render wrong font
![1](https://cloud.githubusercontent.com/assets/2177532/14413230/336347d0-ffa6-11e5-833f-1d07e32e35d7.jpg)

**Comment by unknownbrackets** [comment]
Oh, sorry, just read the history again - this is definitely render-to-texture related somehow, and probably also impacted by expiring FBOs too early.  Hmm.

-[Unknown]

**Comment by unknownbrackets** [comment]
Apparently the JPCSP issue is a flushing issue (which we probably also would share, if we were even rendering properly.)

The renders the labels to a temporary texture, and sets a stall address.  It then updates the source texture it's using to render the labels, and switches to the next render target for the next label.

The problem is, when drawing is deferred until state change, the game changing texture RAM doesn't cause flushing.  So it draws the next text (with confused UVs) to the target.  That's why "OPTION" and "OPTION" twice above, and "CONTINUE" looks funky (since it wants to write "NEW GAME".)

-[Unknown]

**Comment by hrydgard** [comment]
Ugh, that's such a terrible misuse of hardware :)  Ruins parallelism between the CPU and GPU.

So to fix this, we simply need to flush our collected vertices on every stall. Pretty sure that will slow down some games (especially games with large textures, causing more hashing work), though the really high-demanding games usually are smarter than stalling that often so hopefully wouldn't be affected too badly.

**Comment by LunaMoo** [comment]
Tested that a bit and unfortunately flushing in UpdateStall didn't helped much since nothing shows unless Read framebuffers to memory is used or another frame download hack get's added(with different address per each menu option and some menus have lots more options, largest menu I saw had like 10 positions with fb_address starting from 0x001F3A80, up to 0x001FE7C0).

 Using just a new frame download hack it would look alike jpcsp:
![evangelionjo0](https://cloud.githubusercontent.com/assets/5485237/26070482/252142c8-39a5-11e7-93ec-1ec8b2c0eda8.jpg)
 with addition of flush on UpdateStall, while better, it's still messed up:
![evangelionjo1](https://cloud.githubusercontent.com/assets/5485237/26070296/67abc04c-39a4-11e7-9332-097464c3e79f.jpg)

 The messup is also slightly different for different menu's like here seen right after loading game:
![evangelionjo2](https://cloud.githubusercontent.com/assets/5485237/26070302/6f1132d6-39a4-11e7-9bab-8d9747d6a8ef.jpg)
 which should look like:
![evangelionjo3](https://cloud.githubusercontent.com/assets/5485237/26070307/74969d7c-39a4-11e7-8d59-73e84557df3c.jpg)

 Last field as can be seen always ends up outdated ~ taken from previous menu / or empty in case it's first menu. Leaving and re-entering makes that field show correctly since previous menu is then same as current.

 Evil game.:]

**Comment by unknownbrackets** [comment]
Hm.  I don't think UpdateStall is the right place.

Basically the "stall" process works like this (on a real PSP):

1. Game writes code to render some, and enqueues a list starting at A with a stall at B.
2. The GPU starts running - meanwhile, the CPU keeps going.
3. Game writes more code after the stall position B until C, and maybe even updates texture data if it thinks the draw "probably finished by now".
4. Game issues UpdateStall for C - if the GPU reached B, this unpauses it.  Either way, the new stall is C.

So flushing at UpdateStall is too late - that's when the game has already written the new data.  Flushing needs to be done when PPSSPP hits the stall and pauses.

Note: PPSSPP (without multithreading) pauses the CPU until rendering is done (i.e. it's never possible to catch it before it hits B.)  With multithreading, it does more like the above - but the timing is not consistent, and doesn't at all match a real PSP.

-[Unknown]

**Comment by LunaMoo** [comment]
Thanks for that explanation - I set it to flush earlier in InterpretList in place it sets GPUSTATE_STALL now and combined with frame download hack most of the menu works completely fine:
![evangelionjo4](https://cloud.githubusercontent.com/assets/5485237/26141164/c6b67c8a-3ada-11e7-8253-fc8c8be34f41.jpg)
 However the last entry still has same problem, shows only outdated texture from previous menu:]. Might be a different problem?

**Comment by unknownbrackets** [comment]
Do either #11322 or #11323 help?  It seems like they may from some data I saw, but not sure if it still requires an increased old age.

If it does, a few ideas:
 * Is there a way to detect that this game is using a safe size?
 * Perhaps we extend the life of small FBOs (a log tells me they are 12 pixels tall), since they are less likely to overlap with later RAM usage?  But this is still risky...
 * Are there any CPU instructions messing with this VRAM?

-[Unknown]

**Comment by benderscruffy01** [comment]
v1.8.0-42-g62928e13d
still the same problem
[DUMP.ZIP](https://github.com/hrydgard/ppsspp/files/3021545/DUMP.ZIP)

**Comment by unknownbrackets** [comment]
I wonder is this has the same issue as #10967 seems to, as noted above (in the JPCSP bug discussion.)

-[Unknown]

**Comment by Gamemulatorer** [comment]
Using ppsspp latest git build.
Both HW and SW rendering suffer the issue.
![Screenshot_20200604-171055](https://user-images.githubusercontent.com/37603562/83738152-8119c680-a686-11ea-9ead-84a704e31a07.png)
![Screenshot_20200604-171106](https://user-images.githubusercontent.com/37603562/83738163-837c2080-a686-11ea-8655-232d08b24e06.png)

**Comment by nassau-tk** [comment]
v1.10.3-1308 "Buffere Rendering & Simulate Blocktransfer ON"
OPTION's "O" appeared.
![image](https://user-images.githubusercontent.com/48179091/101635198-f8cd1180-3a6c-11eb-9862-dc65c73e8e30.png)

v1.10.3-1308 "NonBuffere Rendering & Simulate Blocktransfer ON"
"O" is missing...
![image](https://user-images.githubusercontent.com/48179091/101635379-3cc01680-3a6d-11eb-8d1a-3fc2e7d95c6a.png)

[ULJS00201_0001.zip](https://github.com/hrydgard/ppsspp/files/5666175/ULJS00201_0001.zip)

**Comment by sum2012** [comment]
The font "O" is fix by https://github.com/hrydgard/ppsspp/commit/f1a3ba7f5669204a964e4a49c0354d8ab64e6dbc
hehe ,this is scefont font !?
But I use real psp of jpn.pgf  ,"O" disappear

**Comment by nassau-tk** [comment]
@sum2012 
Hahaha~!!:stuck_out_tongue_closed_eyes:

Yes,That's right.
PSP Original PGF font is missing any character kind of this loop menu.
But,Kind of other menu (fixed position) is maybe no problem both.
And,I don't have clearly conviction that this game uses PGF font.
If "l" or "g" appear on the menu then I can check about "Is it PGF or not?".

[ Loop menu ]
![image](https://user-images.githubusercontent.com/48179091/102709337-0bf1a400-42ed-11eb-81d0-cd8a503e123f.png)
![image](https://user-images.githubusercontent.com/48179091/102709511-a3a3c200-42ee-11eb-94bb-7d2db4ebeb05.png)

[ Fix position menu ]
![image](https://user-images.githubusercontent.com/48179091/102709423-dac5a380-42ed-11eb-8c03-a07dee134c1d.png)

**Comment by sum2012** [comment]
Small finding.If I disable black list 	"sceFont_Library" and "SceFont_Library",
32:28:756 user_main    I[LOADER]: HLE\sceKernelModule.cpp:1332 Module sceFont_Library: 09ef7de0 09eef594 09eef5c0
32:28:756 user_main    I[LOADER]: HLE\sceKernelModule.cpp:784 Function (sceReg,0cae832b) unresolved in 'sceFont_Library', storing for later resolving
32:28:756 user_main    I[LOADER]: HLE\sceKernelModule.cpp:784 Function (sceReg,1d8a762e) unresolved in 'sceFont_Library', storing for later resolving
32:28:757 user_main    I[LOADER]: HLE\sceKernelModule.cpp:784 Function (sceReg,28a8e98a) unresolved in 'sceFont_Library', storing for later resolving
32:28:757 user_main    I[LOADER]: HLE\sceKernelModule.cpp:784 Function (sceReg,92e41280) unresolved in 'sceFont_Library', storing for later resolving
32:28:757 user_main    I[LOADER]: HLE\sceKernelModule.cpp:784 Function (sceReg,d4475aa8) unresolved in 'sceFont_Library', storing for later resolving
32:28:757 user_main    I[LOADER]: HLE\sceKernelModule.cpp:784 Function (sceReg,fa8a5739) unresolved in 'sceFont_Library', storing for later resolving
32:28:769 user_main    I[LOADER]: HLE\sceKernelModule.cpp:1441 Exporting ent 0 named sceFont_Library, 0 funcs, 2 vars, resident 09eef644
32:28:769 user_main    I[LOADER]: HLE\sceKernelModule.cpp:1441 Exporting ent 1 named sceLibFont, 28 funcs, 0 vars, resident 09eef660
32:28:769 user_main    W[LOADER]: HLE\sceKernelModule.cpp:795 Ignoring func export sceLibFont/02d7f94b, already implemented in HLE.
32:28:769 user_main    W[LOADER]: HLE\sceKernelModule.cpp:795 Ignoring func export sceLibFont/099ef33c, already implemented in HLE.
32:28:769 user_main    W[LOADER]: HLE\sceKernelModule.cpp:795 Ignoring func export sceLibFont/0da7535e, already implemented in HLE.
32:28:769 user_main    W[LOADER]: HLE\sceKernelModule.cpp:795 Ignoring func export sceLibFont/27f6e642, already implemented in HLE.
32:28:769 user_main    W[LOADER]: HLE\sceKernelModule.cpp:795 Ignoring func export sceLibFont/2f67356a, already implemented in HLE.
32:28:769 user_main    W[LOADER]: HLE\sceKernelModule.cpp:795 Ignoring func export sceLibFont/3aea8cb6, already implemented in HLE.
32:28:769 user_main    W[LOADER]: HLE\sceKernelModule.cpp:795 Ignoring func export sceLibFont/3c4b7e82, already implemented in HLE.
32:28:769 user_main    W[LOADER]: HLE\sceKernelModule.cpp:795 Ignoring func export sceLibFont/472694cd, already implemented in HLE.
32:28:769 user_main    W[LOADER]: HLE\sceKernelModule.cpp:795 Ignoring func export sceLibFont/48293280, already implemented in HLE.
32:28:769 user_main    W[LOADER]: HLE\sceKernelModule.cpp:795 Ignoring func export sceLibFont/48b06520, already implemented in HLE.
32:28:769 user_main    W[LOADER]: HLE\sceKernelModule.cpp:795 Ignoring func export sceLibFont/5333322d, already implemented in HLE.
32:28:769 user_main    W[LOADER]: HLE\sceKernelModule.cpp:795 Ignoring func export sceLibFont/568be516, already implemented in HLE.
32:28:769 user_main    W[LOADER]: HLE\sceKernelModule.cpp:795 Ignoring func export sceLibFont/574b6fbc, already implemented in HLE.
32:28:769 user_main    W[LOADER]: HLE\sceKernelModule.cpp:795 Ignoring func export sceLibFont/57fcb733, already implemented in HLE.
32:28:769 user_main    W[LOADER]: HLE\sceKernelModule.cpp:795 Ignoring func export sceLibFont/5c3e4a9e, already implemented in HLE.
32:28:769 user_main    W[LOADER]: HLE\sceKernelModule.cpp:795 Ignoring func export sceLibFont/5dcf6858, already implemented in HLE.
32:28:769 user_main    W[LOADER]: HLE\sceKernelModule.cpp:795 Ignoring func export sceLibFont/67f17ed7, already implemented in HLE.
32:28:769 user_main    W[LOADER]: HLE\sceKernelModule.cpp:795 Ignoring func export sceLibFont/681e61a7, already implemented in HLE.
32:28:769 user_main    W[LOADER]: HLE\sceKernelModule.cpp:795 Ignoring func export sceLibFont/74b21701, already implemented in HLE.
32:28:770 user_main    W[LOADER]: HLE\sceKernelModule.cpp:795 Ignoring func export sceLibFont/980f4895, already implemented in HLE.
32:28:770 user_main    W[LOADER]: HLE\sceKernelModule.cpp:795 Ignoring func export sceLibFont/a834319d, already implemented in HLE.
32:28:770 user_main    W[LOADER]: HLE\sceKernelModule.cpp:795 Ignoring func export sceLibFont/aa3de7b5, already implemented in HLE.
32:28:770 user_main    W[LOADER]: HLE\sceKernelModule.cpp:795 Ignoring func export sceLibFont/bb8e7fe6, already implemented in HLE.
32:28:770 user_main    W[LOADER]: HLE\sceKernelModule.cpp:795 Ignoring func export sceLibFont/bc75d85b, already implemented in HLE.
32:28:770 user_main    W[LOADER]: HLE\sceKernelModule.cpp:795 Ignoring func export sceLibFont/ca1e6945, already implemented in HLE.
32:28:770 user_main    W[LOADER]: HLE\sceKernelModule.cpp:795 Ignoring func export sceLibFont/dcc80c2f, already implemented in HLE.
32:28:770 user_main    W[LOADER]: HLE\sceKernelModule.cpp:795 Ignoring func export sceLibFont/ee232411, already implemented in HLE.
32:28:770 user_main    W[LOADER]: HLE\sceKernelModule.cpp:795 Ignoring func export sceLibFont/f8f0752e, already implemented in HLE.
32:28:770 user_main    I[SCEMODULE]: HLE\sceKernelModule.cpp:2040 296=sceKernelLoadModule(name=disc0:/PSP_GAME/USRDIR/FONT/LIBFONT.PRX,flag=00000000,(...))

Which mean that it use custom scefont library

**Comment by benderscruffy** [comment]
v1.13.1-289-ga35d26190
still missing the words
renders properly in software renderer
here is a GE Dump from the software renderer
[ULJS00201.zip](https://github.com/hrydgard/ppsspp/files/9335456/ULJS00201.zip)

**Comment by unknownbrackets** [comment]
The dump works okay on hardware too, which only confirms the previous findings.  It also makes me think that the stall updating issue may be resolved now.

It sounds like there's just one issue remaining.  My assumption is the game does this:
1. Creates three temporary render surfaces.
2. Draws "NEW GAME", "CONTINUE", and "OPTION" to these temporary surfaces using rendering.
3. Waits a while, animating in the menu or something.
4. Finally uses those pre-rendered surfaces as textures to draw the button text on the actual buttons.

In the software renderer, temporary surfaces are stored in RAM (just like on the PSP.)  This means they're permanent (as long as the game is running), they're always updated by CPU activity, and they are never accidentally overwritten by unrelated rendering.

The hardware renderer, however, creates actual temporary surfaces on your GPU.  They don't exist in RAM, and moreover... it's hard to tell what's safe to update in RAM.  Some types of updates from the CPU don't update them, so if one isn't used for a while we start thinking it must be old and wrong (if we didn't do this, characters would be walking around with "NEW GAME" instead of faces, because we didn't notice it loading the face with the CPU.)

So - presumably - if we made temporary surfaces immortal (hello NEW GAME faces), it would "fix" this bug.  That would support my assumptions above, but wouldn't truly be a way to fix this issue.

Another thing we tried to do was determine a "safe size" to update in RAM.  For complex reasons, we don't even know what part of RAM was actually updated by drawing in the hardware renderers.  If we blindly update RAM after rendering, it makes games crash because we delete important data.  So we watch for signals for what RAM should be updated.  It seems like this game isn't doing any of the things we normally watch for to figure that out.

The most helpful frame dump now would be a frame dump for when it actually draws the temporary surfaces.  It's definitely before you see the "NEW GAME" on screen, but you would see if while stepping through the GE debugger.  However, this would be hard to capture because by the time you see it in the GE debugger, you're already one frame too late to create a frame dump of it.  It's only drawing those temporary surfaces once (if my assumptions are correct.)

Another, simpler thing that could be happening is that we could be misdetecting the texturing as coming from an offset into the wrong surface.  Or it could be rendering these all together to a single temporary surface, but then texturing from offsets, and we could be missing that.

Just wanted to collect all the information about the issue here.

-[Unknown]

**Comment by IrfanH495** [comment]
I tried to bring up the missing text using textures but the result is that in any menu 
1 New game 
2 continue
![Screenshot_20230225-083552_PPSSPP](https://user-images.githubusercontent.com/78854859/221328970-ddaf9182-c0d3-4393-8d41-a8b2bd2646da.png)
![Screenshot_20230225-083654_PPSSPP](https://user-images.githubusercontent.com/78854859/221328973-0e55b7f8-f49a-4761-9486-e1204bf082d0.png)
![Screenshot_20230225-083712_PPSSPP](https://user-images.githubusercontent.com/78854859/221328974-1e9ab2d0-b27d-4562-b5a0-e9ac49beb49d.png)
![Screenshot_20230225-083719_PPSSPP](https://user-images.githubusercontent.com/78854859/221328976-5b5e5956-dd7b-481b-9935-373d4f29f605.png)
![Screenshot_20230225-083732_PPSSPP](https://user-images.githubusercontent.com/78854859/221328977-0bb37d49-358e-46d0-a49a-492d30a6937b.png)
did i make a mistake while making it texture

**Comment by nassau-tk** [comment]
Probably, It will relate PGF issue too.
Because, Shpe of "I" was make by me.

**Comment by IrfanH495** [comment]
2xpsp
vulkan
frame skipping 1

https://user-images.githubusercontent.com/78854859/221393888-87fc619d-6408-401d-a69a-1b68cd96c00d.mp4

**Comment by unknownbrackets** [comment]
Unfortunately, this game is doing something very "anti-frameskip", in that it's drawing some graphics only once and reusing them.  Frameskip skips entire frames for speed, and if it skips one of these "key" frames where one-time drawing occurs, there will be problems.

Unfortunately, the game doesn't tell us whether the drawing is one time or not so we can't tell until after we've already skipped it...

As far as saving as a PNG, does it help to change the texture hash method in textures.ini?

-[Unknown]

**Comment by IrfanH495** [comment]
I've tried it but still the texture doesn't appear
I've also tried installing textures taken from software rendering, but they don't appear.
texture from software rendering
![continue texture software rendering](https://user-images.githubusercontent.com/78854859/221399423-2966568a-61bf-4c94-9487-b57a28127df7.png)
texture vulkan
![continue](https://user-images.githubusercontent.com/78854859/221399485-390d8db7-3790-48b9-a695-4d48ab0f906c.png)

**Comment by unknownbrackets** [comment]
Well, it's probably not a texture like that.

The game is drawing CONTINUE to a framebuffer and using that as a texture.  You can't replace dynamically rendered framebuffers in texture packs.  Instead, you'd need to replace whatever texture it initially uses when drawing CONTINUE.  I assume that's something more like a grid of letters.

-[Unknown]

**Comment by sum2012** [comment]
I ask AI, it answered "The issue was in the PSP font rendering system (Core/Font/PGF.cpp) where:

EVA fonts use very low alpha values (1-15 range) that become invisible during rendering
The original pixel enhancement logic didn't provide sufficient visibility boost for these fonts
Font pixels with very low alpha values were being written but remained practically invisible" but it doesn't give perfect solve ( I tested 16 times) 
code change:  https://github.com/hrydgard/ppsspp/commit/fe104ded3f95f41eea284f90f4493c65b65b65c1
<img width="718" height="443" alt="Image" src="https://github.com/user-attachments/assets/d67b79c5-2133-4e10-a831-1b84eee5e9fa" />

**Comment by sum2012** [comment]
Second try best, will test further before pull request
https://github.com/hrydgard/ppsspp/commit/7b7567da6733ea681f864d710e016eba7d40c63a

**Comment by hrydgard** [comment]
This was fixed in [#21678](https://github.com/hrydgard/ppsspp/pull/21678) a while back. Closing.
